mypy/build.py

```diff
@@ -778,59 +787,73 @@ def is_file(path: str) -> bool:
     return os.path.isfile(path)
 
 
-def find_module(id: str, lib_path_arg: Iterable[str]) -> Optional[str]:
-    """Return the path of the module source file, or None if not found."""
-    lib_path = tuple(lib_path_arg)
+class ModuleDiscovery:
+    def __init__(self, lib_path: Iterable[str], namespaces_allowed: bool):
+        self.lib_path = tuple(lib_path)
+        self.namespaces_allowed = namespaces_allowed
 
-    def find() -> Optional[str]:
+    def find(self, id: str) -> Optional[str]:
```

Comment:
Given that this class now handles both regular modules and namespaces, but this method handles only module-finding internals, I think it'd be clearer to name it `_find_module` instead of `find`; `find` as a public method (and first in source order) strongly implies it's the primary public API, which it's not.